sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormat.scala

```diff
@@ -241,47 +260,74 @@ object FileFormat {
     FileSourceConstantMetadataStructField(FILE_BLOCK_LENGTH, LongType, nullable = false),
     FileSourceConstantMetadataStructField(FILE_MODIFICATION_TIME, TimestampType, nullable = false))
 
+  /**
+   * All [[BASE_METADATA_FIELDS]] require custom extractors because they are derived directly from
+   * fields of the [[PartitionedFile]], and do have entries in the file's metadata map.
```

Comment:
```suggestion
   * fields of the [[PartitionedFile]], and don't have entries in the file's metadata map.
```

Reply:
@ryan-johnson-databricks do I understand this correctly?

Reply:
Erm, yes... I think I was trying to type "do not" but didn't quite make it.